File: VeroApp/Services/ValidationService.swift
import Foundation

enum ValidationService {
    static func isValidEmail(_ email: String) -> Bool {
        let regex = "[A-Z0-9a-z._%+-]+@[A-Za-z0-9.-]+\\.[A-Za-z]{2,64}"
        return NSPredicate(format: "SELF MATCHES %@", regex).evaluate(with: email)
    }

    static func isValidCPF(_ cpf: String) -> Bool {
        let numbers = cpf.filter(\.isNumber)
        guard numbers.count == 11, Set(numbers).count != 1 else { return false }
        let digits = numbers.compactMap { Int(String($0)) }

        func check(_ len: Int) -> Bool {
            let sum = (0..<len).reduce(0) { $0 + digits[$1] * (len + 1 - $1) }
            let rem = sum % 11
            return digits[len] == (rem < 2 ? 0 : 11 - rem)
        }
        return check(9) && check(10)
    }

    static func isValidPhone(_ phone: String) -> Bool {
        let n = phone.filter(\.isNumber).count
        return n >= 10 && n <= 11
    }

    static func isValidCEP(_ cep: String) -> Bool {
        cep.filter(\.isNumber).count == 8
    }
}

File: VeroFlutter/lib/services/validation_service.dart
abstract class ValidationService {
  static bool isValidEmail(String email) =>
      RegExp(r'^[a-zA-Z0-9._%+\-]+@[a-zA-Z0-9.\-]+\.[a-zA-Z]{2,}$')
          .hasMatch(email);

  static bool isValidCPF(String cpf) {
    final d = cpf.replaceAll(RegExp(r'\D'), '');
    if (d.length != 11 || d.split('').toSet().length == 1) return false;

    int calc(int len) {
      int sum = 0;
      for (int i = 0; i < len; i++) sum += int.parse(d[i]) * (len + 1 - i);
      final rem = sum % 11;
      return rem < 2 ? 0 : 11 - rem;
    }

    return int.parse(d[9]) == calc(9) && int.parse(d[10]) == calc(10);
  }

  static bool isValidPhone(String phone) {
    final n = phone.replaceAll(RegExp(r'\D'), '').length;
    return n >= 10 && n <= 11;
  }
}
